### PR TITLE
chore(types): more specific badge types

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Issue

It seems like we where declaring a type we where not expected but this did not fail type checking because we where not specific enough.

## Description

Adds the more specific type and use that where applicable.

### Double check

- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: web/src/components/Badge.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
+export type PillTypes = 'default' | 'warning';
```
Comment (silkeholmebonnen) [inline web/src/components/Badge.tsx:1]: Small thing but would it be more correct to call it `PillType`?

**Review by silkeholmebonnen — APPROVED** [review]
Looks good to me! Just a small nitpick

File: web/src/components/Badge.tsx (hunk 1 above)
Comment (VIKTORVAV99) [reply web/src/components/Badge.tsx:1]: I'm actually not sure what the convention is but I was thinking that the property `pillType` should be a type of `PillTypes` if that makes sense?

I'd be happy to change it if you wish though.

Comment (silkeholmebonnen) [reply web/src/components/Badge.tsx:1]: I did a little google search and it seems that the convention is singular, so `PillType` ([Types in typescript](https://imgeorgemor.medium.com/you-dont-know-how-to-write-types-in-typescript-d8cfaf4857e5))

Comment (VIKTORVAV99) [reply web/src/components/Badge.tsx:1]: Great!

I'll change it then!

**Pull request merged**